activestorage/app/models/active_storage/preview.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+# Some non-image blobs can be previewed: that is, they can be presented as images. A video blob can be previewed by
+# extracting its first frame, and a PDF blob can be previewed by extracting its first page.
+#
+# Active Storage provides previewers for videos and PDFs. You can add your own previewers or remove the built-in ones
+# by modifying +ActiveStorage.previewers+.
+#
+#   ActiveStorage.previewers
+#   # => [ ActiveStorage::Previewer::PdfPreviewer, ActiveStorage::Previewer::VideoPreviewer ]
+#
+#   ActiveStorage.previewers << CustomPreviewer
+#
+# The built-in previewers rely on third-party system libraries:
+#
+#   * {ffmpeg}(https://www.ffmpeg.org)
+#   * {mupdf}(https://mupdf.com)
+#
+# These libraries are not provided by Rails; to use the built-in previewers, you must install them yourself.
```

Comment:
Are we detecting their presence early enough and raising helpful messages? It looks like this is currently setting people up for an unexpected error in production

Reply:
I think it's fine, because they'll fail as soon as you actually call #preview. Which will happen in test/dev. The upside of this is that it's a no config.